Remove NoInlining/StackCrawlMarks from Tasks

stephentoub opened the pull request:
These crawl marks in Tasks aren't relevant in coreclr (they're passed down to ExecutionContext, which in coreclr just ignores them), yet they required NoInlining, which does impact coreclr.  Removing them all  from tasks.

cc: @jkotas, @kouvel

jkotas commented:
Delete the method that takes StackCrawlMark from ExecutionContext?

LGTM otherwise.

benaadams commented:
Has conflicts

stephentoub commented:
> Has conflicts

Yes, I'm working on this.

stephentoub commented:
> Delete the method that takes StackCrawlMark from ExecutionContext?

It was still being used by some other calls in Timer, Overlapped, ThreadPool, and Thread, so I went through and cleaned up those as well, and deleted this Capture overload.  As long as I was doing so, I also cleaned up a few other things I noticed along the way.
